### PR TITLE
Move to Solr 8.7, Biosolr 2.0.0 and auth

### PR DESCRIPTION
**Pull request by pcm32**
This PR:

- Moves Solr to 8.7 (and adequate ZK version)
- Introduces Solr authentication, with an admin user and a query/guest user, modifying all calls.
- Uses BioSolr 2.0.0 (and the mumbo jumbo to add it to Solr 8.7).

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -18,6 +18,8 @@ docker run --rm -i --net mynet -v $( pwd )/tests:/usr/local/tests \
   -e SOLR_HOST=$SOLR_HOST -e ZK_HOST=$ZK_HOST -e ZK_PORT=$ZK_PORT \
   -e POSTGRES_USER=$POSTGRES_USER \
   -e POSTGRES_PASSWORD=$POSTGRES_PASSWORD \
+  -e SOLR_USER=<an-existing-solr-username-that-can-write-or-read> \
+  -e SOLR_PASS=<passsword-forl-that-solr-user> \
```
Comment (alfonsomunozpomer) [inline README.md]: Typo here.

File: README.md [hunk 2]
```diff
@@ -27,6 +29,67 @@ Scripts to create and load data into the `gxa-*` Solr indexes (for analytics). E
...
+
+`bulk-analytics-v1` makes use of the [BioSolr plugin](https://github.com/ebi-gene-expression-group/BioSolr) to perform ontology expansion on document indexing based on EFO. In order to enable BioSolr, there are 2 options:
+
+### Option 1: Local `.jar` file
+
+Place BioSolr jar (which can be found in the repository's `lib` directory) under `/server/solr/lib/` in your Solr installation directory. This is the oldest option, and has some security issues, but for testing should be fine.
+
+### Option 2: Blob store API
+
+You can use the BioSolr jar as a runtime library stored in the blob store API. In order to enable the use of runtime libraries, you must start your Solr instance with the flag `-Denable.runtime.lib=true`. **This option is now deprecated in solr 8 and will not be available anymore in Solr 9.**
```
Comment (alfonsomunozpomer) [inline README.md:65]: Is `-Denable.runtime.lib=true` enabled by default in Solr 8?

File: README.md [hunk 3]
```diff
@@ -27,6 +29,67 @@ Scripts to create and load data into the `gxa-*` Solr indexes (for analytics). E
...
+# Authentication
+
+The setup on the CI is made to use authentication with default user and password. The calls assume these settings (solr:SolrRocks), but the user and password can be modified by doing:
+
+```
+export SOLR_USER=<new-user>
+export SOLR_PASS=<new-pass>
+```
+
+To use default auth in a new solr cloud instance, upload `test/security.json` to ZK as shown in the `Setup auth` part of the `run_tests_in_containers.sh`. To setup users in a production setting the script [create-users.sh](bin/create-users.sh) will receive two set of users:
```
Comment (alfonsomunozpomer) [inline README.md]: Typo: “SolrCloud”. I wouldn’t point it out but I’ve seen you’ve gone the extra mile to write it correctly elsewhere. :smile: Grep and replace “solr cloud”, as it appears in some other bits of the doc.

File: README.md (hunk 2 above)
Comment (pcm32) [reply README.md:65]: Not that I'm aware of, but it is no longer needed when using the packages system (or it is set to true by default). It is not mentioned in the documentation for that usage and all works on the tests without setting it to true.

File: README.md (hunk 1 above)
Comment (pcm32) [reply README.md]: hawk eye.

**Review by alfonsomunozpomer — APPROVED** [review]
All good. I left a couple very minor comments about typos.

**Comment by pcm32** [comment]
Thanks for the review @alfonsomunozpomer . Will tests these branches for loading into our tests and dev instances in Harlow and merge if all is good.

**Comment by pcm32** [comment]
This has successfully loaded into test and dev in HL. It is ready for merging.

**Review by pmb59 — APPROVED** [review]
LGTM

**Comment by pcm32** [comment]
We have established that this is working well on Harlow in the new wwwdev env, so I will merge it now.

**Pull request merged**